### PR TITLE
Wc 27040 new hook needed get query for stock

### PR DESCRIPTION
**Pull request by mfs-mindsize**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

The purpose of this PR is to address the request of:

Issue #27040 - "New hook needed on \WC_Product_Data_Store_CPT::get_query_for_stock()"

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

I'm not sure how to best state this. Sorry.

### CODE REVIEW

**Comment by smachi** [comment]
Hi @mfs-mindsize, 

Thanks a lot for the changes.
Both commits would fit our needs.

But in the commit 8435c5d(https://github.com/woocommerce/woocommerce/pull/28937/commits/8435c5d44a3258b27e36487aeb012fbddbfbb756), I have realized that the **$result** variable may be undefined if **$do_query** is false and no ReserveStockException is triggered.

**Comment by mfs-mindsize** [comment]
@smachi - Yes, I can see that now. Good catch. Thanks. So would you say that this would be better. I ask because I'm betting you're more familiar with the implications than I might be currently. 

```
$result = true;
if ( $do_query ) {
}
```

File: includes/data-stores/class-wc-product-data-store-cpt.php [hunk 1]
```diff
@@ -2111,13 +2111,15 @@ protected function get_primary_key_for_lookup_table( $table ) {
...
+		$query = $wpdb->prepare(
 			"
 			SELECT COALESCE ( MAX( meta_value ), 0 ) FROM $wpdb->postmeta as meta_table
 			WHERE meta_table.meta_key = '_stock'
 			AND meta_table.post_id = %d
 			",
 			$product_id
 		);
+
+		return apply_filters( 'woocommerce_query_for_stock', $query, $product_id );
```
Comment (ObliviousHarmony) [inline includes/data-stores/class-wc-product-data-store-cpt.php:2123]: The name of this query is likely too broad given how specific its implementation is.

**Review by ObliviousHarmony — COMMENTED** [review]
Thanks for the contribution @mfs-mindsize!

One of the things we're iterating on is how to structure the extensibility of the platform such that it won't impede progress due to our commitment to backwards compatibility. The potential issue here is that these filters lock in a commitment that the `ReserveStock` class pass both the insertion and the selection query to filters along with the specific arguments currently used in the stock reservation class.

That isn't to say this won't be something we're interested in allowing just that I think we might want to finish making some of those decisions before pressing forward with more granular filters of this kind.

**Comment by Konamiman** [comment]
Hi @mfs-mindsize, this one is quite old, are you still interested in seeing this merged?

Also I was wondering why the `woocommerce_query_reserve_stock_for_product` filter is a true/false flag for doing the query instead of working as `woocommerce_query_for_stock` (so filtering the entire query).

**Comment by peterfabian** [comment]
Since there are some open questions/unaddressed feedback and not much interest from the community to push this forward, I'm going to close this one. We can revisit/adopt the solution if the enhancement gets more traction. Thanks all for understanding.

**Pull request closed without merging**